lib/puppet/util.rb

```diff
@@ -25,6 +25,67 @@ module Util
 
   extend Puppet::Util::SymbolicFileMode
 
+  public
+  def self.get_environment_variable(name, mode = nil)
+    if mode.nil?
+      if Puppet.features.microsoft_windows?
+        mode = :windows
+      else
+        mode = :posix
+      end
+    end
+
+    if mode == :windows
+      Puppet::Util::Windows::Process.get_environment_strings.each do |key, value |
+        if name.casecmp(key) == 0 then
+          return value
+        end
+      end
+      return nil
+    else
+      ENV[name]
+    end
+  end
+  # TODO use alias_method instead
+  def self.env(name,mode = nil)
+    get_environment_variable(name,mode)
+  end
+
+  def self.get_environment_as_hash(mode = nil)
+    if mode.nil?
+      if Puppet.features.microsoft_windows?
+        mode = :windows
+      else
+        mode = :posix
+      end
+    end
+    get_environment(mode)
+  end
+
+  def self.clear_environment
+    ENV.clear
```

Comment:
Yes it works on windows.  It is called in the spec_helper and in fact that was the only reason I implemented platform specific implementation